pinterest/teletraan
# Implementation of blessed instance types

Background:
InfraGov is rolling out Blessed Instances Types. They are the preferred shortlist of types that most teams should use for their services. The Teletraan UI will introduce friction when selecting non-blessed instance types.

Test Plan:
1. Navigate to clouds/hosttype page
2. Create a new host type and confirm that it has a default value of ONBOARDING for the blessed status 
![image](https://github.com/pinterest/teletraan/assets/69278532/ba4458ce-d010-4169-a619-8a2834c9dfb0)

3. Navigate to the Cluster Configuration page
4. Confirm that the new host type is selectable 
![image](https://github.com/pinterest/teletraan/assets/69278532/1b47221d-518c-4b71-b6d2-c000bbd57b68)

5. Navigate the to cluster creation page
6. Confirm that the new host type is selectable 
![image](https://github.com/pinterest/teletraan/assets/69278532/42ffe001-04b6-49ae-8e75-b200171560e6)
7. Navigate to clouds/hosttype page
8. Modify the new host type by updating the blessed status to SUSTAINING 
![image](https://github.com/pinterest/teletraan/assets/69278532/e6e6d09c-3228-4db5-bf6a-fb9fe117073b)

9. Repeat steps 3-6 confirming that the the host type is selectable for cluster configuration but not cluster creation 
![image](https://github.com/pinterest/teletraan/assets/69278532/68b6eac4-0a8f-430f-8abb-2abe08291bd4)
![image](https://github.com/pinterest/teletraan/assets/69278532/1a4cfb31-0792-4706-9e71-278b986548cb)

10. Update the cluster to use the host type with blessed status of SUSTAINING 
![image](https://github.com/pinterest/teletraan/assets/69278532/b7bf03cf-4455-45a7-8c77-f36e835356b7)

11. Navigate to clouds/hosttype page
12. Update the host type to DECOMMISSIONING 
![image](https://github.com/pinterest/teletraan/assets/69278532/6fd46ae8-86da-44ec-ae7c-221c6a8106af)

13. Navigate the the Environment page and confirm you see a warning directing you to update the host type 
![image](https://github.com/pinterest/teletraan/assets/69278532/cf6f6312-b688-46cc-8d6c-c2900a3db782)

14. Navigate to the cluster configuration page and confirm you see a warning directing you to update the host type 
![image](https://github.com/pinterest/teletraan/assets/69278532/1e288972-b244-47fa-9fa3-e2e332b74400)

15. Confirm that other DECOMMISSIONING host types are not selectable for cluster creation 
![image](https://github.com/pinterest/teletraan/assets/69278532/7638d8ab-88c5-46ba-b714-2865f705036b)

16. Navigate to clouds/hosttypes and update a host type by checking the retired checkbox 
![image](https://github.com/pinterest/teletraan/assets/69278532/5fbc1826-77ff-4c8a-909f-2b7629ca723a)

17. Confirm that the host type is not selectable in cluster configuration 
![image](https://github.com/pinterest/teletraan/assets/69278532/fb69059e-77d3-43b3-8789-23e1a41eaf34)
